From fb2ddc8bf89ca9544015a1dc65f15c56e891d7ea Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 27 Aug 2014 08:57:32 -0400
Subject: [redhat] kabi: remove wireless_send_event from whitelist

Message-id: <20140827085731.GA2451@redhat.com>
Patchwork-id: 89262
O-Subject: [RHEL7 PATCH redhat] kabi: remove wireless_send_event from whitelist
Bugzilla:
RH-Acked-by: Jiri Olsa <jolsa@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

wireless_send_event() function declared as part of kabi also declare
various wireless structures like cfg80211_*, wiphy_*, ieee80211_*.
as part of kabi - those structures are used to calculate
wireless_send_event symbol checksum (that process can be seen on
net/wireless/wext-core.symtypes file, which can be generated by
"make net/wireless/wext-core.symtypes").

Wireless structures were never intended to be part of kabi as we do not
include any other wireless functions as part of it. Is not possible
to write external wireless driver that use kernel wireless structures,
use only in kernel wireless_send_event() function and not use any other
kernel cfg80211_*, wiphy_*, ieee80211_* functions. IOW no external
wireless driver that utilize wireless_send_even() symbol as part of
kernel stable kabi can exist.

Instead of adding a lot of __GENKSYMS__  ifdefs to cheat kabi checker
is much better to remove that symbol from kabi whitelist.

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=7851430
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kabi/kabi-module/kabi_ppc64/wireless_send_event b/redhat/kabi/kabi-module/kabi_ppc64/wireless_send_event
index 79c679e..fac1994 100644
--- a/redhat/kabi/kabi-module/kabi_ppc64/wireless_send_event
+++ b/redhat/kabi/kabi-module/kabi_ppc64/wireless_send_event
@@ -1,2 +1,2 @@
-#0-
+#0-1
 0x0cf73c0d wireless_send_event vmlinux EXPORT_SYMBOL
diff --git a/redhat/kabi/kabi-module/kabi_x86_64/wireless_send_event b/redhat/kabi/kabi-module/kabi_x86_64/wireless_send_event
index 00cd5aa..1b87b62 100644
--- a/redhat/kabi/kabi-module/kabi_x86_64/wireless_send_event
+++ b/redhat/kabi/kabi-module/kabi_x86_64/wireless_send_event
@@ -1,2 +1,2 @@
-#0-
+#0-1
 0xf3b8b280 wireless_send_event vmlinux EXPORT_SYMBOL
-- 
1.7.1